### PR TITLE
1. Adding log msgs in HardDeleter. 2. Fixing throttler in store tools

### PR DESCRIPTION
**Pull request by nsivabalan**
1. Adding log msgs in HardDeleter to print whether hard delete is in progress or awaiting after caught up during shutdown
2. Fixing throttler in store tools to throttle based on index entries per sec instead of index files per sec

SLA: 5 mins
Reviewers: @vgkholla 

Built and coding style applied

### CODE REVIEW

File: ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java [hunk 1]
```diff
@@ -51,6 +51,7 @@
   static final long HARD_DELETE_SLEEP_TIME_ON_CAUGHT_UP_MS = 10 * Time.MsPerSec;
 
   final AtomicBoolean enabled = new AtomicBoolean(true);
+  final AtomicBoolean awaitingAfterCaughtUp = new AtomicBoolean(false);
```
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java]: `private`

Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java]: Also this need not be atomic. `volatile` should do

File: ambry-tools/src/main/java/com.github.ambry/store/DumpLogTool.java [hunk 2]
```diff
@@ -80,10 +80,10 @@ public DumpLogTool(VerifiableProperties verifiableProperties, StoreToolsMetrics
     this.clusterMap =
         ((ClusterAgentsFactory) Utils.getObj(clusterMapConfig.clusterMapClusterAgentsFactory, clusterMapConfig,
             hardwareLayoutFilePath, partitionLayoutFilePath)).getClusterMap();
-    if (blobsPerSec > 0) {
-      this.throttler = new Throttler(blobsPerSec, 1000, true, SystemTime.getInstance());
-    } else if (blobsPerSec < 0) {
-      throw new IllegalArgumentException("BlobsPerSec " + blobsPerSec + " cannot be negative ");
+    if (bytesPerSec > 0) {
+      this.throttler = new Throttler(bytesPerSec, 1000, true, SystemTime.getInstance());
+    } else if (bytesPerSec < 0) {
```
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpLogTool.java]: you can ensure this when you load the property by providing a range from 0 to Integer MAX

**Comment by nsivabalan** [comment]
@vgkholla : addressed your comments

**Comment by nsivabalan** [comment]
@vgkholla : Resolved merge conflicts

**Pull request merged**